mlflow/gateway/providers/mlflow.py

```diff
@@ -0,0 +1,136 @@
+from fastapi import HTTPException
+from fastapi.encoders import jsonable_encoder
+
+from .base import BaseProvider
+from .utils import send_request
+from ..config import RouteConfig, MLflowConfig
+from ..schemas import completions, chat, embeddings
+
+
+class MLflowProvider(BaseProvider):
+    def __init__(self, config: RouteConfig) -> None:
+        super().__init__(config)
+        if config.model.config is None or not isinstance(config.model.config, MLflowConfig):
+            raise TypeError(f"Invalid config type {config.model.config}")
+        self.mlflow_config: MLflowConfig = config.model.config
+        self.headers = {"Content-Type": "application/json"}
+
+    @staticmethod
+    def process_payload(payload, key):
+        payload = jsonable_encoder(payload, exclude_none=True)
+
+        # Remove the route-added regulation value from being passed to MLflow hosted models
+        payload.pop("candidate_count", None)
+
+        input_data = payload.pop(key)
+        request_payload = {"inputs": input_data if isinstance(input_data, list) else [input_data]}
+
+        if payload:
+            request_payload["params"] = payload
+
+        return request_payload
+
+    async def completions(self, payload: completions.RequestPayload) -> completions.ResponsePayload:
+        # Example request to MLflow REST API server for completions:
+        # {
+        #     "inputs": ["hi", "hello", "bye"],
+        #     "params": {
+        #         "temperature": 0.5,
+        #         "top_k": 3,
+        #     }
+        # }
+
+        resp = await send_request(
+            headers=self.headers,
+            base_url=self.mlflow_config.mlflow_api_base,
+            path="invocations",
+            payload=self.process_payload(payload, "prompt"),
+        )
+
+        # Example response:
+        # {"predictions": ["hello", "hi", "goodbye"]}
+
+        return completions.ResponsePayload(
+            **{
+                "candidates": [
+                    {"text": resp["predictions"], "metadata": {"finish_reason": "length"}}
```

Comment:
For consistency with Databricks model serving (and the eventual consistency with the signature across gateway, serving, and SoTA models), I think we should expect `resp["predictions"]` to be of the form `{"candidates": ["completion candidate 1", "completion candidate 2", ...]}`.